x-pack/plugins/alerting/common/api_key_functions.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest } from 'kibana/server';
+import {
+  SecurityPluginSetup,
+  GrantAPIKeyResult as SecurityPluginGrantAPIKeyResult,
+  InvalidateAPIKeyResult as SecurityPluginInvalidateAPIKeyResult,
+  InvalidateAPIKeyParams,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../security/server';
+
+export type CreateAPIKeyResult =
+  | { apiKeysEnabled: false }
+  | { apiKeysEnabled: true; result: SecurityPluginGrantAPIKeyResult };
+export type InvalidateAPIKeyResult =
+  | { apiKeysEnabled: false }
+  | { apiKeysEnabled: true; result: SecurityPluginInvalidateAPIKeyResult };
+
+export const createAPIKey = async (
+  request: KibanaRequest,
+  securityPluginSetup: SecurityPluginSetup
```

Comment:
You can add `?` to allow passing `undefined` (undefined handled in the lines 27-29).

I would also change all the places calling this function to not have the `!` suffix for TypeScript.

```suggestion
  securityPluginSetup?: SecurityPluginSetup
```